### PR TITLE
Adding new method to InventoryManager that removes an item from a given ...

### PR DESCRIPTION
**Pull request by MarcinSc**
...slot.

Made IntegerRange implements Iterable<Integer>.
Fixed HorizontalBlockFamily to return a block if invoked without secondary side, as in rare circumstances there might be none.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/52/
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/math/IntegerRange.java [hunk 1]
```diff
@@ -33,6 +33,11 @@ public void addNumbers(int from, int to) {
         }
     }
 
+    @Override
+    public Iterator<Integer> iterator() {
+        return createIterator();
+    }
+
     public Iterator<Integer> createIterator() {
```
Comment (msteiger) [inline engine/src/main/java/org/terasology/math/IntegerRange.java]: Is there a reason for having two identical methods with similar names next to each other? If not, I'd remove `createIterator` and fix the calling code. There shouldn't be too many hits.

**Comment by msteiger** [comment]
It would be great if you could fix the checkstyle issues that were introduced with previous PRs, e.g. `IntegerRange.addNumbers` needs braces.

**Comment by Cervator** [comment]
Yeah I find it useful to hit the "Check modified files" in the Checkstyle tab then fix anything outstanding even if I didn't touch it. Anything not overly complicated anyway :-)

Whenever there is an API change so to say that might need fixes in modules I can check on it, just helps to have it highlighted

Also, help me review/test the launcher kthxplz! :D

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/53/
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/55/

**Build Log**
*last 5 lines*

 ```
[...truncated 2087 lines...]
	at hudson.model.Executor.run(Executor.java:240)
[ANALYSIS-COLLECTOR] Skipping publisher since build result is FAILURE
Notifying upstream projects of job completion
Setting status of 83ac430ccf1bfd7155e7867516dd880746db0516 to FAILURE with url http://jenkins.terasology.org/job/TerasologyPRs/55/ and message: Merged build finished.
```

Uh oh, something went wrong with the build. Need to check on that

**Comment by Cervator** [comment]
Looks like the occasional Artifactory crash was the culprit

Gooey: test this please

**Comment by Cervator** [comment]
test this please

No fair, it works for @msteiger :(

**Comment by Cervator** [comment]
Test please

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/56/
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/57/
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/58/
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Did some general inventory testing both in single, client/server + client, and headless + 2 x clients. Seemed fine. Seems the client disposal issues are getting worse lately though (not this PR - pre-existing) - all the plain clients fail to dispose the game correctly.

@MarcinSc did you mean to get https://github.com/MarcinSc/Terasology/commit/bf94cca17c85eff9931f6c3729f669da317c80f9 rolled into this PR? Topic branches ftw! :-)

**Pull request merged**